### PR TITLE
Get ref genome from meta_study instead of command line

### PR DESCRIPTION
**Pull request by SRodenburg**
The reference genome (and thereby species and build) is now retrieved from the `meta_study.txt` file after the **optional** `reference_genome` tag.

Allowed values are **hg19**, **hg38** and **mm10**. The validator will raise an error when any other value is encountered, and will default to **hg19**.

I also added a more specific error when validating the `NCBI_Build` column in MAF files, validating whether the correct reference genome is used.

This was already present in the FIle Formats documentation.

### CODE REVIEW

File: core/src/main/scripts/importer/cbioportal_common.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 ADD_CASE_LIST_CLASS = "org.mskcc.cbio.portal.scripts.AddCaseList"
 VERSION_UTIL_CLASS = "org.mskcc.cbio.portal.util.VersionUtil"
 
-# provides a key for data types to metafile specification dict.  
+# provides a key for data types to metafile specification dict.
```
Comment (SRodenburg) [inline core/src/main/scripts/importer/cbioportal_common.py:40]: somehow my editor automatically removes trailing whitespace

File: core/src/main/scripts/importer/cbioportal_common.py [hunk 2]
```diff
@@ -685,7 +686,6 @@ def validate_types_and_id(meta_dictionary, logger, filename):
 def parse_metadata_file(filename,
                         logger,
                         study_id=None,
-                        genome_name=None,
```
Comment (SRodenburg) [inline core/src/main/scripts/importer/cbioportal_common.py:688]: somehow this parameter is passed on but never used.

File: core/src/main/scripts/importer/metaImport.py [hunk 3]
```diff
@@ -70,16 +70,7 @@ def interface():
...
-                                            'from the cBioPortal installation')
-    parser.add_argument('-species', '--species', type=str, default='human',
-                        help='species information (default: assumed human)',
-                        required=False)
-    parser.add_argument('-ucsc', '--ucsc_build_name', type=str, default='hg19',
-                        help='UCSC reference genome assembly name (default: assumed hg19)',
-                        required=False)   
-    parser.add_argument('-ncbi', '--ncbi_build_number', type=str, default='37',
-                         help='NCBI reference genome build number (default: assumed 37 for UCSC reference genome build hg19)',
-                         required=False)
```
Comment (SRodenburg) [inline core/src/main/scripts/importer/metaImport.py:82]: arguments became obsolete now

File: core/src/main/scripts/importer/validateData.py [hunk 4]
```diff
@@ -4453,11 +4459,18 @@ def process_metadata_files(directory, portal_instance, logger, relaxed_mode, str
     global meta_dictionary
     tags_file_path = None
 
+    # implemented ref genomes
+    reference_genome_map = {
+        'hg19':('human','37','hg19'),
+        'hg38':('human','38','hg38'),
+        'mm10':('mouse','38','mm10')
+    }
+
```
Comment (SRodenburg) [inline core/src/main/scripts/importer/validateData.py:4468]: not sure if we should also allow **GRCxXX** values

File: core/src/main/scripts/importer/validateData.py [hunk 5]
```diff
@@ -4864,8 +4892,6 @@ def validate_data_relations(validators_by_meta_type, logger):
 def request_from_portal_api(server_url, api_name, logger):
     """Send a request to the portal API and return the decoded JSON object."""
 
-    print(api_name)
-
```
Comment (SRodenburg) [inline core/src/main/scripts/importer/validateData.py:4868]: relic from previous work

**Review by SRodenburg — COMMENTED** [review]
some sidenotes

**Comment by inodb** [comment]
And maybe double check if https://github.com/cBioPortal/cbioportal-docker-compose still works for grch38. Thanks!

**Comment by SRodenburg** [comment]
> And maybe double check if https://github.com/cBioPortal/cbioportal-docker-compose still works for grch38. Thanks!

Just tested, it still works. I did not change anything there.

File: core/src/main/scripts/importer/validateData.py [hunk 6]
```diff
@@ -4518,6 +4531,21 @@ def process_metadata_files(directory, portal_instance, logger, relaxed_mode, str
...
+                                 list(reference_genome_map.keys()),
+                                 extra={
+                                     'filename_':filename,
+                                     'cause':meta_dictionary['reference_genome'].strip()
+                                 })
+                else:
+                    portal_instance.species, portal_instance.ncbi_build, portal_instance.genome_build = \
+                        reference_genome_map[meta_dictionary['reference_genome']]
+            else:
+                logger.info('No reference genome specified -- using default (hg19)',
```
[the remark this answers is not in the document]
Comment (SRodenburg) [reply core/src/main/scripts/importer/validateData.py:4546]: The `portal.properties` are not used in the validator. AFAIK, those are only used in the Java code, and some parameters are passed on to the frontend.

[the remark this answers is not in the document]
Comment (oplantalech) [reply core/src/main/scripts/importer/validateData.py:4546]: @SRodenburg I see. I was confused by the `portal_instance` name.

Defaults for these properties are, indeed, set here: https://github.com/cBioPortal/cbioportal/pull/7969/files/e805ff2f295eae71032d52dfcd754849a07a401c#diff-9be14d6564536099d9c431cf330dc2ad62abfecbbc191ac7c5814855dc37685eL305 

So I think this three values can be simplified into one. As I mentioned, hg19/hg38/mm10 already gives you all the information given from the three properties that we currently have, and there is no need to deviate from the `reference_genome` property

**Comment by SRodenburg** [comment]
@inodb docs are updated :heavy_check_mark:

**Review by oplantalech — APPROVED** [review]

File: docs/Support-multiple-reference-genomes.md [hunk 7]
```diff
@@ -20,20 +20,4 @@ if your study is profiled by a reference geome other than a default genome value
...
-                        NCBI reference genome build number (default: assumed
-                        GRCh37 for UCSC reference genome build hg19)
-```
-For instance, to load a hg38 dataset:
-```
-core/src/main/scripts/importer/metaImport.py  -s /path/to/your/study -jar scripts.jar -ucsc hg38 -ncbi GRCh38 -n -o
-
-```
-
+* You can also overwrite default genome values by using the following options of the importer script. For that, you have to specify the `reference_genome` field in the `meta_study.txt` file of your study.
```
Comment (inodb) [inline docs/Support-multiple-reference-genomes.md:23]: not entirely clear here what the values are supposed to be but will update that

**Pull request merged**